easybuild/easyblocks/m/motioncor2.py

```diff
@@ -0,0 +1,117 @@
+##
+# Copyright 2019-2019 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing MotionCor2, implemented as an easyblock
+
+@author: Ake Sandgren, (HPC2N, Umea University)
+"""
+
+import os
+import stat
+
+from easybuild.framework.easyblock import EasyBlock
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.filetools import adjust_permissions, copy_file, mkdir, write_file
+from easybuild.tools.modules import get_software_root
+
+
+class EB_MotionCor2(EasyBlock):
+    """
+    Support for installing MotionCor2
+     - creates wrapper that loads the correct version of CUDA before
+     - running the actual binary
+    """
+
+    def prepare_step(self, *args, **kwargs):
+        super(EB_MotionCor2, self).prepare_step(*args, **kwargs)
+
+        if not get_software_root('CUDA'):
+            raise EasyBuildError("CUDA must be a direct (build)dependency of MotionCor2")
+
+        self.cuda_mod_name, self.cuda_name = None, None
+        self.motioncor2_bin = None
```

Comment:
@akesandgren Sorry for the nitpicking, but the initialization of class variables belongs in the constructor (`__init__`). :)

Can you move these two lines to `__init__`?

```python
def __init__(self, *args, **kwargs):
    """Constructor of MotionCor2 easyblock."""
    super(EB_MotionCor2, self).__init__(*args, **kwargs)

    self.cuda_mod_name, self.cuda_name = None, None
    self.motioncor2_bin = None